### PR TITLE
Use new solver during selection

### PR DESCRIPTION
**Pull request by compiler-errors**
r? @lcnr

### CODE REVIEW

**Comment by compiler-errors** [comment]
Oh cool, so it must've just been noise when I tested previously.

**Comment by lcnr** [comment]
cool

@bors r+ rollup=never

File: compiler/rustc_trait_selection/src/traits/select/mod.rs [hunk 1]
```diff
@@ -586,18 +594,40 @@ impl<'cx, 'tcx> SelectionContext<'cx, 'tcx> {
...
         }
-        Ok(result)
+    }
+
+    /// Evaluates the predicates using the new solver when `-Ztrait-solver=next` is enabled
+    fn evaluate_predicates_recursively_in_new_solver(
+        &mut self,
+        predicates: impl IntoIterator<Item = PredicateObligation<'tcx>>,
+    ) -> Result<EvaluationResult, OverflowError> {
+        let mut fulfill_cx = FulfillmentCtxt::new();
```
Comment (lcnr) [inline compiler/rustc_trait_selection/src/traits/select/mod.rs:620]: can you instead use the following here. It feels a bit confusing to simply use a fulfillment context here without specifying which one.

```suggestion
        let mut fulfill_cx = crate::solve::FulfillmentCtxt::new();
```

**Comment by compiler-errors** [comment]
Fixed to use full path

@bors r=lcnr

**Pull request merged**